### PR TITLE
GSYE-140: Deprecate multiple orders matching

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

File: src/gsy_e/models/market/two_sided.py [hunk 1]
```diff
@@ -371,36 +354,31 @@ def _validate_requirements_satisfied(
 
     @classmethod
     def validate_bid_offer_match(
-            cls, bids: List[Bid], offers: List[Offer],
+            cls, bid: Bid, offer: Offer,
             clearing_rate: float, selected_energy: float) -> None:
         """Basic validation function for bids against offers.
```
Comment (faizan2590) [inline src/gsy_e/models/market/two_sided.py:359]:
```suggestion
        """Basic validation function for bid against offer.
```

File: src/gsy_e/models/market/two_sided.py [hunk 2]
```diff
@@ -320,36 +317,22 @@ def match_recommendations(
...
+            trade_bid_info = TradeBidOfferInfo(
+                original_bid_rate=original_bid_rate,
+                propagated_bid_rate=market_bid.energy_rate,
+                original_offer_rate=market_offer.original_price / market_offer.energy,
+                propagated_offer_rate=market_offer.energy_rate,
+                trade_rate=original_bid_rate)
+
+            bid_trade, offer_trade = self.accept_bid_offer_pair(
+                market_bid, market_offer, clearing_rate,
+                trade_bid_info, min(selected_energy, market_offer.energy, market_bid.energy))
```
Comment (faizan2590) [inline src/gsy_e/models/market/two_sided.py:330]: Minor but better to have an small variable to keep the selected_energy
```
selected_energy=min(selected_energy, market_offer.energy, market_bid.energy)
```

```suggestion
                trade_bid_info, selected_energy)
```

**Review by faizan2590 — APPROVED** [review]
Some minor comment, other than that LGTM

File: src/gsy_e/models/market/two_sided.py (hunk 2 above)
Comment (Hassan754) [reply src/gsy_e/models/market/two_sided.py:330]: I think that we should avoid creating variables for values that aren't reused, pylint will complain about this too

**Review by faizan2590 — APPROVED** [review]

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**